rustls/src/ticketer.rs

```diff
@@ -148,17 +148,36 @@ impl TicketSwitcher {
     ///
     /// Calling this regularly will ensure timely key erasure.  Otherwise,
     /// key erasure will be delayed until the next encrypt/decrypt call.
-    fn maybe_roll(
-        &self,
-        now: TimeBase,
-        state: &mut MutexGuard<TicketSwitcherState>,
-    ) -> Result<(), rand::GetRandomFailed> {
+    ///
+    /// In order to reduce the ammount of lock/unlock operations in the
+    /// common case that no new ticketer is needed, we return the mutex-
+    /// guard taken here
```

Comment:
It would be good to explain some of the stuff that Brian mentioned in his original ticket here, which seems like a fairly different/more specific motivation than what you mention here.

Reply:
hmmm, after thinking and rereading brians original comment, i realized the solution here is not ideal either, I will spend some time soon to get to a good solution that covers all his goals